ocaml/xapi/xapi_vbd_helpers.ml

```diff
@@ -169,12 +169,21 @@ let valid_operations ~expensive_sharing_checks ~__context record _ref' : table =
     if not vdi_record.Db_actions.vDI_managed
     then set_errors Api_errors.vdi_not_managed [ _ref ] all_ops;
 
-    let vdi_operations_besides_copy =
+    let vdi_operations_cannot_be_performed_live =
+      let illegal_operation = function
+        | `snapshot
+        | `resize_online
+        | `blocked
+        | `clone
+        | `mirror
+        | `copy -> false
+        | _     -> true
+      in
```

Comment:
The thing about copy is that it's a very long running operation that doesn't involve the SM backends (except right at the beginning and end). Mirror is very similar. The other operations you've listed here don't have that property.

I don't think they should all be added to the 'whitelist' since the behaviour in that case would be that 'other_operation_in_progress' would be raised, which would trigger a retry after a short delay, and hence the operation would likely succeed anyway.

You've also renamed this refering to 'live' operations - but I don't believe that's really the intent. The point is more to restrict concurrent operations on the VDIs where this would cause problems in the SM layer (although there are further checks done there too, but they would more likely cause a permanent failure rather than the backoff-retry behaviour of 'other_operations_in_progress').